sync/class.jetpack-sync-client.php

```diff
@@ -0,0 +1,672 @@
+
+		if ( ! $buffer ) {
+			// buffer has no items
+			return false;
+		}
+
+		if ( is_wp_error( $buffer ) ) {
+			// another buffer is currently sending
+			return false;
+		}
+
+		$upload_size   = 0;
+		$items_to_send = array();
+
+		// we estimate the total encoded size as we go by encoding each item individually
+		// this is expensive, but the only way to really know :/
+		foreach ( $buffer->get_items() as $key => $item ) {
+
+			// expand item data, e.g. IDs into posts (for full sync)
+			$item[1] = apply_filters( "jetpack_sync_before_send_" . $item[0], $item[1] );
+
+			$encoded_item = $this->codec->encode( $item );
+			$upload_size += strlen( $encoded_item );
+
+			if ( $upload_size > $this->upload_max_bytes && count( $items_to_send ) > 0 ) {
+				break;
+			}
+
+			$items_to_send[ $key ] = $encoded_item;
+		}
+
+		/**
+		 * Fires when data is ready to send to the server.
+		 * Return false or WP_Error to abort the sync (e.g. if there's an error)
+		 * The items will be automatically re-sent later
+		 *
+		 * @since 4.1
+		 *
+		 * @param array $data The action buffer
+		 */
+
+		$result = apply_filters( 'jetpack_sync_client_send_data', $items_to_send );
+
+		if ( ! $result || is_wp_error( $result ) ) {
+			// error_log("There was an error sending data:");
+			// error_log(print_r($result, 1));
+			$result = $this->sync_queue->checkin( $buffer );
+
+			if ( is_wp_error( $result ) ) {
+				error_log( "Error checking in buffer: " . $result->get_error_message() );
+				$this->sync_queue->force_checkin();
+			}
+			// try again in 1 minute
+			$this->schedule_sync( "+1 minute" );
+		} else {
+
+			// scan the sent data to see if a full sync started or finished
+			if ( $this->buffer_includes_action( $buffer, 'jetpack_full_sync_start' ) ) {
+				$this->full_sync_client->set_status_sending_started();
+			}
+
+			if ( $this->buffer_includes_action( $buffer, 'jetpack_full_sync_end' ) ) {
+				$this->full_sync_client->set_status_sending_finished();
+			}
+
+			$this->sync_queue->close( $buffer, $result );
+			// check if there are any more events in the buffer
+			// if so, schedule a cron job to happen soon
+			if ( $this->sync_queue->has_any_items() ) {
+				$this->schedule_sync( "+1 minute" );
+			}
+		}
+	}
+
+	private function buffer_includes_action( $buffer, $action_name ) {
+		foreach ( $buffer->get_items() as $item ) {
+			if ( $item[0] === $action_name ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	function expand_wp_insert_post( $args ) {
+		// list( $post_ID, $post, $update ) = $args;
+		return array( $args[0], $this->filter_post_content_and_add_links( $args[1] ), $args[2] );
+	}
+
+	// Expands wp_insert_post to include filteredpl
+	function filter_post_content_and_add_links( $post ) {
+		if ( 0 < strlen( $post->post_password ) ) {
+			$post->post_password = 'auto-' . wp_generate_password( 10, false );
+		}
+		$post->post_content_filtered = apply_filters( 'the_content', $post->post_content );
```

Comment:
Could you document this filter? You can simply refer to the original filter, like so:
`/** This filter is already documented in core. wp-includes/post-template.php */`